extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/test/CalciteMSQTestsHelper.java

```diff
@@ -416,6 +416,9 @@ protected static Supplier<ResourceHolder<CompleteSegment>> getSupplierForSegment
       case CalciteTests.T_ALL_TYPE_PARQUET:
         index = TestDataBuilder.getQueryableIndexForDrillDatasource(segmentId.getDataSource(), tempFolderProducer.apply("tmpDir"));
         break;
+      case CalciteTests.BENCHMARK_DATASOURCE:
```

Comment:
Is there scope in reusing this datasource for anything else? If so, it could be renamed to a more general name. If not, I'm wondering if there is a better way to override this just for the one test.

Reply:
> Is there scope in reusing this datasource for anything else? If so, it could be renamed to a more general name.

No, there isn't any scope in re-using this for anything else.

> If not, I'm wondering if there is a better way to override this just for the one test.

I'm not sure if that's doable, it's not trivial to achieve atleast.

With that said, we did it in a similar way when we added datasources for drill test suite (the `case` statements just above the added line of code) - so if we really want to, I feel it could be a follow-up patch for both drill and benchmark datasources. But I also think it doesn't add any value to over-engineer this, since this change doesn't affect any other test. Hope that makes sense!

Reply:
It does look a bit strange to have details of a datasource used by a single test here. This class would probably explode in size if this was the case for more tests.  We can refactor it in a followup though.